### PR TITLE
Remove mongoose model exports for AuditModel and WatchModel and refactor

### PR DESCRIPTION
**Pull request by msamper**
### Features and Changes
This PR moves us closer to having no mongoose model exports to prevent developers from interacting with our models directly. In this PR I specifically worked on the following models:
- AuditModel
- WatchModel

- Makes progress on #1300 

### Testing
PR should be no-op aside from some new error handling added for `getAllHistory` and `getHistory` within the organizations controller which map to `/history/:type` and `/history/:type/:id` respectively. A type that is not within `EntityType` should now return an HTTP 400 response with an explanation of what the valid types are.

### CODE REVIEW

File: packages/back-end/src/routers/organizations/organizations.controller.ts [hunk 1]
```diff
@@ -96,6 +93,13 @@ import {
   getExperimentsForActivityFeed,
 } from "../../models/ExperimentModel";
 import { removeEnvironmentFromSlackIntegration } from "../../models/SlackIntegrationModel";
+import {
+  findAllByEntityType,
+  findAllByEntityTypeParent,
+  findByEntity,
+  findByEntityParent,
```
Comment (tinahollygb) [inline packages/back-end/src/routers/organizations/organizations.controller.ts:100]: Can we add the word "audit" in these method names? I imagine if every model defined these generic methods that we would have import collisions and would need to remap. In our other models we use the resource name, e.g. `createProject` or `findProjectById` etc.

File: packages/back-end/src/models/WatchModel.ts [hunk 2]
```diff
@@ -11,4 +12,76 @@ watchSchema.index({ userId: 1, organization: 1 }, { unique: true });
...
+      upsert: true,
+    }
+  );
+}
+
+export async function deleteWatchedByEntity(
+  organization: string,
+  userId: string,
+  type: string,
+  id: string
```
Comment (tinahollygb) [inline packages/back-end/src/models/WatchModel.ts:73]: Can we use an options object? With 4 positional string arguments, we are bound to make a mistake.

File: packages/back-end/src/models/AuditModel.ts [hunk 3]
```diff
@@ -34,7 +37,120 @@ const auditSchema = new mongoose.Schema({
...
+
+/**
+ * Convert the Mongo document to an AuditInterface, omitting Mongo default fields __v, _id
+ * @param doc
+ */
+const toInterface = (doc: AuditDocument): AuditInterface => {
+  return omit(doc.toJSON<AuditDocument>(), ["__v", "_id"]);
+};
+
+export async function insertAudit(data: Partial<AuditInterface>) {
```
Comment (tinahollygb) [inline packages/back-end/src/models/AuditModel.ts:52]: We are trying to be explicit with the return types, can you add the function return type?

**Review by tinahollygb — COMMENTED** [review]
Left some comments. I don't know if you saw it yet but these style concerns are referenced in [our style guide](https://www.notion.so/growthbook/Style-Guide-a6943db980214b03bf17e8b021e882f8), specifically being explicit with TypeScript types and not relying on type inference, as well as using options objects instead of positional arguments for functions with several parameters.

**Review by jdorn — APPROVED** [review]

**Pull request merged**